dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

```diff
@@ -1298,6 +1300,72 @@ public void shouldRenderUsingOtherSiteContainerAndNotAdvanceTemplate() throws Ex
         Assert.assertTrue(html.contains("content2content1"));
     }
 
+    //Data Provider for the Widget Pre-execute code test
+    @DataProvider
+    public static Object[] dataProviderWidgetPreExecuteCode() {
+        return new Object[] {
+                new WidgetPreExecuteCodeTestCase(PageMode.EDIT_MODE),
+                new WidgetPreExecuteCodeTestCase(PageMode.PREVIEW_MODE),
+                new WidgetPreExecuteCodeTestCase(PageMode.LIVE),
+        };
+    }
+
+    private static class WidgetPreExecuteCodeTestCase {
+        final PageMode pageMode;
+
+        public WidgetPreExecuteCodeTestCase(final PageMode pageMode) {
+            this.pageMode = pageMode;
+        }
+    }
+
+    @Test
+    @UseDataProvider("dataProviderWidgetPreExecuteCode")
+    public void test_WidgetPreExecuteCodeShowRegardlessPageMode(final WidgetPreExecuteCodeTestCase testCase) throws Exception {
+
+        //Update the Preexecute Field to have some code in it
+        final String preExcuteCode = "PreExecute Code Displayed";
+        ContentType contentType = TestDataUtils.getWidgetLikeContentType();
+        final Field preExecuteField = APILocator.getContentTypeFieldAPI().byContentTypeIdAndVar(contentType.id(),WidgetContentType.WIDGET_PRE_EXECUTE_FIELD_VAR);
+        APILocator.getContentTypeFieldAPI()
+                .save(FieldBuilder.builder(preExecuteField).values(preExcuteCode).build(), systemUser);
+        // Assert that the widget has set the pre-execute field
+        Assert.assertTrue(APILocator.getContentTypeFieldAPI()
+                .byContentTypeIdAndVar(contentType.id(),WidgetContentType.WIDGET_PRE_EXECUTE_FIELD_VAR)
+                .values().equals(preExcuteCode));
+
+        //Create Contentlet
+        final Contentlet widgetContentlet = TestDataUtils.getWidgetContent(true,1,contentType.id());
+        addAnonymousPermissions(widgetContentlet);
+
+        //Create Container, Template and Page
+        final Container container = createContainer();
+        final Template template = createTemplate(container);
+
+        final String pageName = "testPage-"+System.currentTimeMillis();
+        final HTMLPageAsset page = createHtmlPageAsset(template, pageName, 1);
+
+        //Add contentlet to the page
+        MultiTree multiTree = new MultiTree(page.getIdentifier(), container.getIdentifier(),widgetContentlet.getIdentifier() ,UUID,0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'multiTree' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5178716)